### PR TITLE
Simpler top-level cmakelists

### PR DESCRIPTION
**Pull request by doingnz**
## Description
Removed most of the custom RTOS & target logic as the approach encouraged additional script for each additional target. 
This PR does not change the logic for the two ESP targets other than to use the same named CMake variables for the RTOS and Target folders to show how close it is to being the same as the other targets. Given the ESP targets are being reworked for the update to 4.x, I have left that change to the person doing the ESP update to remove ESP difference.

## Motivation and Context
Ideally there would be no RTOS or TARGET specific logic in this top-level CMakeLists.txt to aim for reduced responsibility and remove chance of unexpected interactions when new platforms added. its purpose should be restricted to check a target exists and validate high level shared options common to all nanoFramework targets. It also manages the selection of the standard target or community target. 

With this change, a new ${RTOS} & ${TARGET} will be found simply by creation of the folders under /targets. i.e. /targets/${RTOS}/${TARGET}

## How Has This Been Tested?
Confirmed can build ESP32 and STM32 targets. 

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -64,51 +64,16 @@ if(NOT RTOS OR RTOS STREQUAL "")
...
-set(RTOS_SUPPORTED "ChibiOS" "FreeRTOS" "FreeRTOS_ESP32" "TI_SimpleLink" CACHE INTERNAL "supported RTOS options")
+# check the RTOS and TARGET folders exist. 
+# optionally add custom logic in the last section of this CMakeLists.txt 
+# to perform custom add_subdirectory() actions.
 
-list(FIND RTOS_SUPPORTED ${RTOS} RTOS_INDEX)
-if(RTOS_INDEX EQUAL -1)
-    message(FATAL_ERROR "\n\n'${RTOS}' is an invalid option for RTOS. Please choose a valid RTOS.\n\n")
+if(EXISTS ${CMAKE_SOURCE_DIR}/targets/${RTOS})
+    message(STATUS "\nSetting RTOS to ${RTOS} Exists!\n")
```
Comment (josesimoes) [inline CMakeLists.txt:72]:
```suggestion
    message(STATUS "\nSetting RTOS to ${RTOS}\n")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -774,45 +739,13 @@ endif()
...
-        endif()
-
-    endif()
+# Define PLATFORM base path
+set(BASE_PATH_FOR_PLATFORM ${CMAKE_CURRENT_SOURCE_DIR})
 
 #######################
 # FreeRTOS
-elseif(RTOS_FREERTOS_CHECK)
+if(RTOS_FREERTOS_CHECK)
```
Comment (josesimoes) [inline CMakeLists.txt:748]: I guess this should be:
```suggestion
if(RTOS_FREERTOS_ESP32_CHECK)
```

**Review by josesimoes — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (doingnz) [reply CMakeLists.txt:748]: There are two platforms at the start of the "custom" platforms logic related to FreeRTOS. Please see a couple of lines below your suggested edit the if(RTOS_FREERTOS_ESP32_CHECK)

```
#######################
# FreeRTOS
if(RTOS_FREERTOS_CHECK)

    add_subdirectory(targets/FreeRTOS)

#######################
# FreeRTOS ESP32
elseif(RTOS_FREERTOS_ESP32_CHECK)
```

Comment (josesimoes) [reply CMakeLists.txt:748]: If this is the IF for FREERTOS, then the code below doesn't belong here as it's validating RTC for ESP32, which would belong to _another_ IF....

Comment (josesimoes) [reply CMakeLists.txt:748]: Forget my comment above. I was mislead by the partial representatiuon of the diff on git. After looking at the complete file, I can see the code block is correct.

**Review by josesimoes — APPROVED** [review]
LGTM!

**Comment by nightlark** [comment]
@josesimoes I'm not sure how nfbot is deciding when to apply the `invalid` label, since it seems to be getting added to merged PRs and ones that have been closed and replaced by a newer PR. Given that hacktoberfest is apparently counting `invalid` labels towards disqualification, use of the label by nfbot should probably be paused until after October.

**Comment by josesimoes** [comment]
> @josesimoes I'm not sure how nfbot is deciding when to apply the `invalid` label, since it seems to be getting added to merged PRs and ones that have been closed and replaced by a newer PR. Given that hacktoberfest is apparently counting `invalid` labels towards disqualification, use of the label by nfbot should probably be paused until after October.

@nightlark criteria for adding invalid label is for PRs that where closed without being merged. This one must have been a glitch resulting from the order the git events where processed. The code processing these hasn't been changed for months now. I've also randomly checked several other PRs and couldn't find any that was wrongly labeled. 
Thanks for sharing your concern! 😃 👍🏻

**Pull request merged**